platforms/cpu/src/CpuKernels.cpp

```diff
@@ -166,6 +166,12 @@ class CpuCalcForcesAndEnergyKernel::InitForceTask : public ThreadPool::Task {
                 posq[4*i+1] = (float) posData[i][1];
                 posq[4*i+2] = (float) posData[i][2];
             }
+        
+        // Check for invalid positions.
+        
+        for (int i = 4*start; i < 4*end; i++)
+            if (posq[i] != posq[i])
```

Comment:
FWIW, without c99 or c++11, here are the macro-hoops that numpy goes through to define a cross-platform isnan: https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/npy_math.h#L160-L203